src/main/java/frc/robot/subsystems/swerve/gyro/pigeon2/Pigeon2Gyro.java

```diff
@@ -23,9 +23,11 @@ public class Pigeon2Gyro implements ISwerveGyro {
     private final Pigeon2 gyro;
     private final StatusSignal<Double> yawSignal, xAccelerationSignal, yAccelerationSignal, zAccelerationSignal;
     private final Queue<Double> yawQueue, timestampQueue;
+    private final SwerveGyroConstants constants;
 
-    public Pigeon2Gyro(CTREDeviceID gyroID, Pigeon2Configuration configuration){
+    public Pigeon2Gyro(CTREDeviceID gyroID, Pigeon2Configuration configuration, String swerveLogPath){
         this.gyro = new Pigeon2Wrapper(gyroID);
+        this.constants = new SwerveGyroConstants(swerveLogPath);
```

Comment:
Get log path, don't create const object just for the log path.
Also don't save the alert log path do alertPath + logPath (only with alerts)